runtime/common/src/fees.rs

```diff
@@ -263,14 +267,20 @@ macro_rules! impl_market_creator_fees {
         }
 
         impl MarketCreatorFee {
+            fn do_get_fee(market_id: MarketId, amount: Balance) -> Result<Balance, DispatchError> {
+                let market = MarketCommons::market(&market_id)?; // Should never fail
+                let fee_amount = market.creator_fee.mul_floor(amount);
+                Ok(fee_amount)
+            }
+
```

Comment:
> Are you satisfied if we use unwrap_or_default() in both instances? Balance implements the default trait using Zero.

Seems fine to me. After all, the function always return the amount of fees deducted. If the transfer fails, no fees change hands, so the function should return zero.

> I don't think your suggestion solves the problem what needs to happen if do_distribute errors.

What's the problem?

Reply:
Ah, so the problem is that `get_fee` returns the wrong value if `distribute` errors. That's a good argument for always using the return value of `distribute` instead of `get_fee`. If you want to implement `get_fee` correctly, I guess you need to ensure that the transfer will actually go through or return zero.

Reply:
You can cherry-pick from this branch: https://github.com/zeitgeistpm/zeitgeist/tree/mkl-orderbook-abstraction

I'd say `get_fee` must either be rewritten as described above _or_ market as only to be used for testing purposes.

Reply:
102a742142560efc91718c967f01c95b95cfbdc7